### PR TITLE
Add variants/jump-traced.jl

### LINKED ISSUE #44: Add function tracing variant of JuMP
To show that the macros don't do anything special.

### PR DESCRIPTION
**Pull request by odow**
Closes #44

This is based on the `optimization.jl` model. The purpose is to show that there is nothing special about JuMP's macro, and that the problem can be traded and efficient derivatives constructed.

### CODE REVIEW

**Comment by odow** [comment]
Can wait for #52 so we have tests

**Review by ccoffrin — APPROVED** [review]
The docs at the top of this file could probably use some more discussion of why this is implemented like this. So that a new person finding rosetta-opf JuMP user does not accidentally think this is a recommended way to write a JuMP model.

**Comment by odow** [comment]
How's this now?

File: variants/jump-traced.jl [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+        "time_data" => data_load_time,
+        "time_build" => model_build_time,
+        "time_solve" => solve_time,
+        "time_callbacks" => total_callback_time,
+        "solution" => Dict(k => sol[v] for (k, v) in var_lookup),
+    )
+end
+
+if isinteractive() == false
+    solve_opf("$(@__DIR__)/data/pglib_opf_case5_pjm.m")
```
Comment (odow) [inline variants/jump-traced.jl:402]:
```suggestion
    solve_opf("$(@__DIR__)/../data/pglib_opf_case5_pjm.m")
```

**Pull request merged**